lib/game.rb

```diff
@@ -0,0 +1,99 @@
+require 'pry'
+require_relative 'player.rb'
+require_relative 'board.rb'
+require_relative 'players/computer.rb'
+require_relative 'players/human.rb'
+
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  def game_type(input)
+    case input
+    when "0"
+      @player_1 = Players::Computer.new("X")
+      @player_2 = Players::Computer.new("O")
+    when "1"
+      @player_1 = Players::Human.new("X")
+      @player_2 = Players::Computer.new("O")
+    when "2"
+      @player_1 = Players::Human.new("X")
+      @player_2 = Players::Human.new("O")
+    end
+  end
+
+
+  def current_player
+    @board.turn_count % 2 == 0 ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw?
+  end
+
+ def won?
+   matches = WIN_COMBINATIONS.select do |combo|
+    tokens = combo.map {|cell| @board.cells[cell]}
+    tokens.uniq.size == 1 && tokens.none? { |token| token == " " }
+   end
+
+   if matches.size > 1
+     @board.cells[matches[0][0]] == player_1.token ? matches[0] : matches[1]
+   else
+     matches[0]
+   end
+ end
```

Comment:
Here, another option is to use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```